### PR TITLE
Fix the needle dir when worker has cache diretory

### PR DESCRIPTION
**Pull request by Amrysliu**
Relate ticket: https://progress.opensuse.org/issues/99378

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
Please put more details in the git commit message details (not just the reference to the ticket). And there is a typo s/diretory/directory/

**Comment by Amrysliu** [comment]
> Please put more details in the git commit message details (not just the reference to the ticket). And there is a typo s/diretory/directory/

Thanks. Added more details in commit message.

**Review by okurz — COMMENTED** [review]
s/diretory/directory/

I did not understand the sentence "This is because we didn't receive the cache directory when count needle_dir." . What do you mean by "count"? Can you rephrase?

**Review by Martchus — COMMENTED** [review]
Right, we should definitely pass through all the parameters. Maybe it makes also sense to just write it as:
```
sub needledir { productdir(@_) . '/needles' }
```

**Comment by Amrysliu** [comment]
> s/diretory/directory/
> 
> I did not understand the sentence "This is because we didn't receive the cache directory when count needle_dir." . What do you mean by "count"? Can you rephrase?

I simplified the commit message. Hope it would be clear. :)

**Review by okurz — COMMENTED** [review]
commit message is fine. As you try to fix an actual issue, can you add a test?

**Review by Martchus — APPROVED** [review]

**Comment by Amrysliu** [comment]
> commit message is fine. As you try to fix an actual issue, can you add a test?



> commit message is fine. As you try to fix an actual issue, can you add a test?

Added

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]
